src/daft-local-execution/src/sources/scan_task.rs

```diff
@@ -21,25 +23,56 @@ impl ScanTaskSource {
     pub fn new(scan_tasks: Vec<Arc<ScanTask>>) -> Self {
         Self { scan_tasks }
     }
+
+    #[instrument(
+        name = "ScanTaskSource::process_scan_task_stream",
+        level = "info",
+        skip_all
+    )]
+    async fn process_scan_task_stream(
+        scan_task: Arc<ScanTask>,
+        sender: SingleSender,
+        morsel_size: usize,
+    ) {
+        let io_stats = IOStatsContext::new("MicroPartition::from_scan_task");
+        let stream_result = tokio::task::spawn_blocking(move || {
+            MicroPartition::from_scan_task_streaming(scan_task, io_stats, morsel_size)
+        })
+        .await
+        .context(JoinSnafu {});
+
+        match stream_result {
+            Ok(Ok(mut stream)) => {
+                while let Some(partition) = stream.next().await {
+                    let _ = sender.send(partition.map(Arc::new)).await;
+                }
+            }
+            Ok(Err(e)) => {
+                let _ = sender.send(Err(e.into())).await;
+            }
+            Err(e) => {
+                let _ = sender.send(Err(e.into())).await;
+            }
+        }
+    }
```

Comment:
i think we can do this without channels or  needing to `spawn_blocking`.

if you change `get_data` below as follows

```rs
    fn get_data(&self) -> SourceStream {
        let morsel_size = get_morsel_size();
        use futures::stream::StreamExt;

        futures::stream::iter(self.scan_tasks.clone())
            .map(move |scan_task| {
                MicroPartition::from_scan_task_streaming(
                    scan_task,
                    IOStatsContext::new("MicroPartition::from_scan_task"),
                    morsel_size,
                )
            })
            .try_flatten()
            .boxed()
    }
```

then change `from_scan_task_streaming` to return `crate::Result<BoxStream<'static, DaftResult<Arc<Self>>>>` instead of `crate::Result<BoxStream<'static, DaftResult<Self>>>`

Reply:
Agree that spawn_blocking is not needed. I kinda used it as a band aid because internally the I/O code uses a separate Tokio runtime. We can avoid this by just using the current runtime of the execution engine instead. 

One question though, I think [try_flatten](https://docs.rs/futures/latest/futures/stream/trait.TryStreamExt.html#method.try_flatten) will initiate and exhaust one stream after the other, but ideally i'd like them all to start at the same time and get buffered in something (channel, vec, etc).